src/ostree/ot-builtin-config.c

```diff
@@ -85,18 +88,32 @@ ostree_builtin_config (int argc, char **argv, OstreeCommandInvocation *invocatio
 
   if (!strcmp (op, "set"))
     {
-      if (argc < 4)
+      if (opt_group && argc < 5)
```

Comment:
Shouldn't this be `argc < 3`?  Oh wait, I see.  OK, I think we want the `--group` flag to take an argument.   Rather than `G_OPTION_ARG_NONE`, use `G_OPTION_ARG_STRING`.  Then the `opt_group` will itself contain the group variable.  Does that make sense?

Doing this should simplify a lot of the processing, particularly if we combine the `argc` validation with parsing, something like:

```
if (opt_group)
  {
    if (argc < 3)
      {
        g_set_error (...)
        goto out;
      }

    section = g_strdup (opt_group);
    key = g_strdup (argv[2]);
    value = g_strdup (argv[3]);
  }
else
  {
    if (argc < 3)
      {
        g_set_error (...)
        goto out;
      }

    section_key = argv[2];
    value = argv[3];
    if (!split_key_string (section_key, &section, &key, error))
      goto out;
  }
```

Reply:
Thanks Colin for reviewing it!
I agree that using G_OPTION_ARG_STRING makes things simpler. I have updated PR with required changes.